scout/adapter/mongo/institute.py

```diff
@@ -121,29 +93,39 @@ def update_institute(
                 if group_abbreviations:
                     abbreviation = group_abbreviations[i]
                 existing_groups[hpo_term] = {"name": description, "abbr": abbreviation}
-            updates["$set"]["phenotype_groups"] = existing_groups
-
-        ADMIN_SETTINGS = {
-            "alamut_key": alamut_key,
-            "alamut_institution": alamut_institution,
-            "clinvar_key": clinvar_key,
-            "show_all_cases_status": show_all_cases_status,
-            "soft_filters": soft_filters,
+
+        UPDATE_SETTINGS = {
+            "cohorts": cohorts,
+            "collaborators": sharing_institutes,
+            "coverage_cutoff": coverage_cutoff,
+            "display_name": display_name,
+            "frequency_cutoff": frequency_cutoff,
+            "gene_panels": gene_panels,
+            "gene_panels_matching": gene_panels_matching,
+            "loqusdb_id": loqusdb_ids,
+            "sanger_recipients": sanger_recipients,
+            "clinvar_submitters": clinvar_submitters,
+            "alamut_key": alamut_key,  # Admin setting
+            "alamut_institution": alamut_institution,  # Admin setting
+            "clinvar_key": clinvar_key,  # Admin setting
+            "phenotype_groups": existing_groups,
+            "show_all_cases_status": show_all_cases_status,  # Admin setting
+            "soft_filters": soft_filters,  # Admin setting
+            "check_show_all_vars": check_show_all_vars is not None,
         }
-        for key, value in ADMIN_SETTINGS.items():
-            if value not in [None, "", []]:
+        for key, value in UPDATE_SETTINGS.items():
+            if bool(value) is True:
                 updates["$set"][key] = value
+            else:
+                updates["$unset"][key] = ""  # Remove the key from the institute document
```

Comment:
Let's see - the whole issue with separating admin from user was that for ordinary users, the keys would not be set on the form, but that would not mean we want them unset in db. But since they are really tested the same way now, we should be fine!